pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -308,11 +309,10 @@ public PulsarService(ServiceConfiguration config,
                 new DefaultThreadFactory("zk-cache-callback"));
 
         if (config.isTransactionCoordinatorEnabled()) {
-            this.transactionReplayExecutor = Executors.newScheduledThreadPool(
-                    config.getNumTransactionReplayThreadPoolSize(),
-                    new DefaultThreadFactory("transaction-replay"));
+            this.internalExecutorService = new ExecutorProvider(this.getConfiguration().getNumIOThreads(),
+                    "pulsar-executorProvider-internal");
```

Comment:
```suggestion
                    "pulsar-transaction-executor");
```

Currently, no other domains use this executor provider, we can just give a name that is used by transactions.